File: Python/Math-Array-String/83.py
# 26. Remove Duplicates from Sorted Array

from typing import List

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums:
            return 0

        # Initialize a pointer for the unique elements
        k = 1  # Start from index 1 since the first element is always unique

        for i in range(1, len(nums)):
            if nums[i] != nums[i - 1]:  # Check if the current number is different from the previous one
                nums[k] = nums[i]  # Move the unique number forward
                k += 1  # Increment the unique counter

        return k
    
sol = Solution()
output = sol.removeDuplicates([1,1,2])
print(output)  # Output: 2

File: JavaScript/Math-Array-String/83.js
// 26. Remove Duplicates from Sorted Array


var removeDuplicates = function(nums) {
    if (nums.length === 0) return 0;

    let k = 1; // Pointer for unique elements

    for (let i = 1; i < nums.length; i++) {
        if (nums[i] !== nums[i - 1]) { // If the current element is different from the previous
            nums[k] = nums[i]; // Move the unique element forward
            k++; // Increase unique count
        }
    }

    return k;
};

let nums = [0,0,1,1,1,2,2,3,3,4];
let k = removeDuplicates(nums);
console.log(k, nums.slice(0, k)); // Output: 5 [0, 1, 2, 3, 4]
